File: src/main/java/net/minecord/beautyindicator/model/Combat.java
package net.minecord.beautyindicator.model;

public class Combat {

    private String nameToRestore;
    private int insertedSeconds;
    private int seconds;

    public Combat(String nameToRestore, int seconds) {
        this.nameToRestore = nameToRestore;
        this.insertedSeconds = seconds;
        this.seconds = seconds;
    }

    public int getSeconds() {
        return seconds;
    }

    public void resetSeconds() {
        seconds = insertedSeconds;
    }

    public String getNameToRestore() {
        return nameToRestore;
    }

    public void doUpdate() {
        seconds--;
    }
}

File: src/main/kotlin/net/minecord/beautyindicator/model/Combat.kt
package net.minecord.beautyindicator.model

class Combat(val nameToRestore: String?, val customNameVisibleToRestore: Boolean, private val insertedSeconds: Int) {
    var seconds: Int = 0
        private set

    init {
        this.seconds = insertedSeconds
    }

    fun resetSeconds() {
        seconds = insertedSeconds
    }

    fun doUpdate() {
        seconds--
    }
}
